packages/core-ethereum/src/index.ts

```diff
@@ -74,14 +74,23 @@ export default class HoprEthereum extends EventEmitter {
       this.options?.maxConfirmations ?? CONFIRMATIONS,
       INDEXER_BLOCK_RANGE
     )
-    this.status = 'CREATING'
-    log('Ready to call singleton wrapper', ChainWrapperSingleton)
-    ChainWrapperSingleton.create(this.options, this.privateKey).then((chain: ChainWrapper) => {
-      // Debug log used in e2e integration tests, please don't change
-      log(`connector chain created`)
-      this.status = 'CREATED'
-      this.chain = chain
-    })
+    // In some cases, we want to make sure the chain within the connector is not triggered
+    // automatically but instead via an event. This is the case for `hoprd`, where we need
+    // to get notified after ther chain was properly created, and we can't get setup the
+    // listeners before the node was actually created.
+    if (automaticChainCreation) {
+      this.createChain()
+    } else {
+      this.on('connector:create', this.createChain)
+    }
+  }
+
+  private async createChain(): Promise<void> {
+    this.chain = await ChainWrapperSingleton.create(this.options, this.privateKey)
+    // Debug log used in e2e integration tests, please don't change
+    log(`connector chain created`)
```

Comment:
Is there a particular reason for not using a plain string?

Reply:
Not sure what you mean, as in a single string w/o spaces? e.g. `connector:chain:created` ? Does it make a difference?

Reply:
I meant why a template string instead of single quotes.

Reply:
If there are no variables inside there's no difference nor it should matter tbh

Reply:
> If there are no variables inside there's no difference nor it should matter tbh

it doesn't matter, that's why I didn't request a change. Just asked out of curiosity.